lib/wise_homex/api_client_impl/creator.ex

```diff
@@ -0,0 +1,103 @@
+defmodule WiseHomex.ApiClientImpl.Creator do
+  @moduledoc """
+  Auto generator for api client implementation functions at compile time.
+  """
+
+  @doc """
+  When a module uses this module, we will inject the autogenerated api client functions.
+  """
+  defmacro __using__([]) do
+    WiseHomex.ApiDefinition.api_definition()
+    |> WiseHomex.ApiDefinition.to_api_resources()
+    |> Enum.map(&WiseHomex.ApiClientImpl.Creator.create_endpoint/1)
+  end
+
+  @doc """
+  Create an index endpoint
+  """
+  def create_endpoint({:index, api_resource}) do
+    function_name = "get_#{api_resource.name_plural}" |> String.to_atom()
+
+    quote do
+      def unquote(function_name)(config, query \\ %{}) do
+        WiseHomex.Request.get(config, unquote(api_resource.path), query)
+      end
+    end
+  end
+
+  @doc """
+  Create a show endpoint
+  """
+  def create_endpoint({:show, api_resource}) do
+    function_name = "get_#{api_resource.name_singular}" |> String.to_atom()
+
+    quote do
+      def unquote(function_name)(config, id, query \\ %{}) do
+        id = "#{unquote(api_resource.path)}/#{id}"
```

Comment:
Did you mean `path = ...`?

Reply:
Yes. Done in https://github.com/wise-home/wise_homex/pull/107/commits/9b2feb9788ae7f23e2158ec57c8e9d5ef1c9ca85